src/types/Graph.ts

```diff
@@ -13,6 +13,13 @@ export interface SummaryData {
   summaryTarget: any;
 }
 
+export enum Protocol {
+  GRPC = 'grpc',
+  HTTP = 'http',
+  TCP = 'tcp'
+}
+export type ProtocolType = Protocol.GRPC | Protocol.HTTP | Protocol.TCP;
```

Comment:
Is `ProtocolType` really necessary?
I think you could use `Protocol` enum to type variables.

Reply:
Oops.  +1, At first I had ProtocolType with the string literals and then I added the enum and basically forgot that I no longer needed the type.